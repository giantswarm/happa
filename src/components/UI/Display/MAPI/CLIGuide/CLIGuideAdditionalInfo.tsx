import { Box, Heading, Text } from 'grommet';
import { normalizeColor } from 'grommet/utils';
import React from 'react';
import styled from 'styled-components';

const StyledLink = styled.a`
  color: ${({ theme }) => normalizeColor('input-highlight', theme)};
  display: inline-block;
  width: auto;
`;

interface ICLIGuideAdditionalInfoLink {
  label: string;
  href: string;
  external?: boolean;
}

interface ICLIGuideAdditionalInfoProps
  extends React.ComponentPropsWithoutRef<'div'> {
  links?: ICLIGuideAdditionalInfoLink[];
}

const CLIGuideAdditionalInfo: React.FC<
  React.PropsWithChildren<ICLIGuideAdditionalInfoProps>
> = ({ links, ...props }) => {
  return (
    <Box {...props}>
      <Box>
        <Heading level={5}>Additional information</Heading>
      </Box>
      <Box direction='column' gap='xxsmall'>
        {links?.map((link) => (
          <div key={`${link.label}-${link.href}`}>
            <StyledLink
              href={link.href}
              rel={link.external ? 'noopener noreferrer' : undefined}
              target={link.external ? '_blank' : undefined}
            >
              {link.external && (
                <Text
                  className='fa fa-open-in-new'
                  aria-hidden={true}
                  role='presentation'
                  aria-label='Opens in a new tab'
                  margin={{ right: 'xsmall' }}
                />
              )}
              {link.label}
            </StyledLink>
          </div>
        ))}
      </Box>
    </Box>
  );
};

export default CLIGuideAdditionalInfo;
