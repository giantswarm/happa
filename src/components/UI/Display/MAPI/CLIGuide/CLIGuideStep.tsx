import { Box, Text } from 'grommet';
import PropTypes from 'prop-types';
import React from 'react';
import { CodeBlock, Prompt } from 'UI/Display/Documentation/CodeBlock';

interface ICLIGuideStepProps
  extends Omit<React.ComponentPropsWithoutRef<typeof Box>, 'title'> {
  title: React.ReactNode;
  command: React.ReactNode;
}

const CLIGuideStep: React.FC<ICLIGuideStepProps> = ({
  title,
  command,
  children,
  ...props
}) => {
  return (
    <Box direction='column' gap='small' {...props}>
      <Text>{title}</Text>
      <CodeBlock>
        <Prompt>{command}</Prompt>
      </CodeBlock>
      {children}
    </Box>
  );
};

CLIGuideStep.propTypes = {
  title: PropTypes.string.isRequired,
  command: PropTypes.string.isRequired,
  children: PropTypes.node,
};

export default CLIGuideStep;
