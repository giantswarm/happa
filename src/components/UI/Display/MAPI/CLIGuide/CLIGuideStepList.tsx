import { Box } from 'grommet';
import React from 'react';

interface ICLIGuideStepListProps
  extends React.ComponentPropsWithoutRef<typeof Box> {}

const CLIGuideStepList: React.FC<ICLIGuideStepListProps> = (props) => {
  return <Box direction='column' gap='medium' {...props} />;
};

CLIGuideStepList.propTypes = {};

export default CLIGuideStepList;
