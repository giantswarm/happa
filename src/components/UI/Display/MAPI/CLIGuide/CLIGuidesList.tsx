import {
  Accordion,
  AccordionPanel,
  Box,
  Text,
  ThemeContext,
  ThemeType,
} from 'grommet';
import { normalizeColor } from 'grommet/utils';
import React, { useState } from 'react';
import { css } from 'styled-components';

const customTheme: ThemeType = {
  icon: {
    // @ts-expect-error
    extend: css`
      fill: ${({ theme }) => normalizeColor('text-weak', theme)};
      stroke: ${({ theme }) => normalizeColor('text-weak', theme)};
    `,
  },
};

interface ICLIGuidesListProps
  extends Omit<React.ComponentPropsWithoutRef<typeof Accordion>, 'title'> {
  title?: string;
}

const CLIGuidesList: React.FC<React.PropsWithChildren<ICLIGuidesListProps>> = ({
  children,
  title = 'Use the Management API to â€¦',
  ...props
}) => {
  const [isExpanded, setIsExpanded] = useState(false);

  return (
    <ThemeContext.Extend value={customTheme}>
      <Accordion
        onActive={(indices: number[]) => setIsExpanded(indices.length > 0)}
        basis='100%'
        round='xsmall'
        background={isExpanded ? 'accent-strong' : 'background-strong'}
        {...props}
      >
        <AccordionPanel
          label={
            <Box
              direction='row'
              align='center'
              pad='small'
              gap='xsmall'
              round='xsmall'
              background={isExpanded ? 'accent-strong' : 'background-strong'}
            >
              <Text
                className='fa fa-shell'
                role='presentation'
                aria-hidden={true}
                color='text-weak'
              />
              <Text weight='bold' color='text-weak'>
                {title}
              </Text>
            </Box>
          }
        >
          <Box
            basis='100%'
            margin={{ bottom: 'xsmall' }}
            pad='12px'
            gap='small'
          >
            {children &&
              React.Children.map(
                children,
                (child) =>
                  child &&
                  React.cloneElement(child as React.ReactElement, {
                    inList: true,
                  })
              )}
          </Box>
        </AccordionPanel>
      </Accordion>
    </ThemeContext.Extend>
  );
};

export default CLIGuidesList;
