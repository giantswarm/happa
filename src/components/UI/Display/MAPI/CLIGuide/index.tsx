import {
  Accordion,
  AccordionPanel,
  Box,
  Text,
  ThemeContext,
  ThemeType,
} from 'grommet';
import { normalizeColor } from 'grommet/utils';
import React from 'react';
import styled, { css } from 'styled-components';

const StyledAccordion = styled(Accordion)`
  code {
    background-color: ${({ theme }) => normalizeColor('accent-strong', theme)};
    font-size: 1em;

    var {
      font-style: normal;
    }
  }
`;

const customTheme: ThemeType = {
  global: {
    colors: {
      text: 'text-weak',
    },
  },
  accordion: {
    border: false,
  },
  icon: {
    // @ts-expect-error
    extend: css`
      fill: ${({ theme }) => normalizeColor('text-weak', theme)};
      stroke: ${({ theme }) => normalizeColor('text-weak', theme)};
    `,
  },
};

interface ICLIGuideProps
  extends Omit<React.ComponentPropsWithoutRef<typeof Accordion>, 'title'> {
  title: React.ReactNode;
  footer?: React.ReactNode;
}

const CLIGuide: React.FC<React.PropsWithChildren<ICLIGuideProps>> = ({
  children,
  title,
  footer,
  ...props
}) => {
  return (
    <ThemeContext.Extend value={customTheme}>
      <StyledAccordion background='background-strong' round='xsmall' {...props}>
        <AccordionPanel
          label={
            <Box
              direction='row'
              gap='xsmall'
              align='center'
              pad={{ vertical: 'small', horizontal: 'small' }}
            >
              <Text
                className='fa fa-shell'
                role='presentation'
                aria-hidden={true}
              />
              <Text weight='bold'>{title}</Text>
            </Box>
          }
        >
          <Box
            pad={{ vertical: 'medium', horizontal: 'large' }}
            width={{ max: 'xlarge' }}
          >
            {children}

            {footer && (
              <Box
                border={{ side: 'top' }}
                pad={{ vertical: 'small' }}
                margin={{ top: 'medium' }}
              >
                {footer}
              </Box>
            )}
          </Box>
        </AccordionPanel>
      </StyledAccordion>
    </ThemeContext.Extend>
  );
};

export default CLIGuide;
