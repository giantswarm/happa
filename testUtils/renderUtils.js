import { render } from '@testing-library/react';
import App from 'App';
import { ConnectedRouter } from 'connected-react-router';
import { ThemeProvider } from 'emotion-theming';
import { createMemoryHistory } from 'history';
import React from 'react';
import { Provider } from 'react-redux';
import { AppRoutes } from 'shared/constants/routes';
import configureStore from 'stores/configureStore';
import theme from 'styles/theme';

export const initialStorage = {
  user:
    '{"email":"developer@giantswarm.io","auth":{"scheme":"giantswarm","token":"a-valid-token"},"isAdmin":true}',
};

/**
 * This function will render the whole app with a mocked store in the route
 * provided.
 * @param {String} route.
 * @param {HTMLElement} container An HTMLElement.
 */
export function renderRouteWithStore(
  initialRoute = AppRoutes.Home,
  state = {},
  storage = initialStorage
) {
  localStorage.replaceWith(storage);

  const history = createMemoryHistory({
    initialEntries: [initialRoute],
    initialIndex: 0,
  });

  const store = configureStore(state, history);

  const app = render(<App {...{ store, theme, history }} />);

  return app;
}

/**
 * Render component in the Theme context
 * @param {React.ReactType} component The React Component
 * @param {<P extends Record<string, any>>React.ComponentProps<P>} props Props to pass to the component
 * @param {<Q extends TestingLibrary.Queries>TestingLibrary.RenderOptions<Q>} [options] Testing library render options
 */
export function renderWithTheme(component, props, options) {
  return render(getComponentWithTheme(component, props), options);
}

/**
 * Include component in the Theme context and return it
 * @param {React.ReactType} Component The React Component
 * @param {<P extends Record<string, any>>React.ComponentProps<P>} props Props to pass to the component
 */
export function getComponentWithTheme(Component, props) {
  return (
    <ThemeProvider theme={theme}>
      <Component {...props} />
    </ThemeProvider>
  );
}

/**
 * Render component in the Store context
 * @param {React.ReactType} component The React Component
 * @param {<P extends Record<string, any>>React.ComponentProps<P>} props Props to pass to the component
 * @param {Record<string, any>} state Current Store state
 * @param {<Q extends TestingLibrary.Queries>TestingLibrary.RenderOptions<Q>} [options] Testing library render options
 */
export function renderWithStore(component, props, state, options) {
  return render(getComponentWithStore(component, props, state), options);
}

/**
 * Include component in the Store context and return it
 * @param {React.ReactType} Component The React Component
 * @param {<P extends Record<string, any>>React.ComponentProps<P>} props Props to pass to the component
 * @param {Record<string, any>} state Current Store state
 * @param {Record<string, any>} storage Current LocalStorage value
 * @param {History} history Current Browser history
 */
export function getComponentWithStore(
  Component,
  props = {},
  state = {},
  storage = initialStorage,
  history = createMemoryHistory()
) {
  localStorage.replaceWith(storage);

  const store = configureStore(state, history);

  const app = (
    <Provider store={store}>
      <ThemeProvider theme={theme}>
        <ConnectedRouter history={history}>
          <Component {...props} />
        </ConnectedRouter>
      </ThemeProvider>
    </Provider>
  );

  return app;
}
