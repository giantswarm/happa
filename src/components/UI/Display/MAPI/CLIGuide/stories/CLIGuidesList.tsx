import { Story } from '@storybook/react';
import React from 'react';

import CLIGuide from '..';
import CLIGuidesList from '../CLIGuidesList';
import CLIGuideStep from '../CLIGuideStep';
import CLIGuideStepList from '../CLIGuideStepList';

export const Simple: Story<
  React.ComponentPropsWithoutRef<typeof CLIGuidesList>
> = (args) => {
  return <CLIGuidesList {...args} />;
};

Simple.args = {
  title: 'Use the API to â€¦',
  children: (
    <CLIGuide title='Get some data from the somewhere'>
      <CLIGuideStepList>
        <CLIGuideStep
          title='1. Make sure you are a boss'
          command='some command --some-flag'
        />
        <CLIGuideStep
          title='2. Maybe check again if you are still a boss'
          command='some-other-cli checkboss'
        />
      </CLIGuideStepList>
    </CLIGuide>
  ),
};

Simple.argTypes = {
  title: { control: { type: 'text' } },
};
