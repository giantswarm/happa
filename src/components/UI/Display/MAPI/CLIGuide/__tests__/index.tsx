import { fireEvent, screen } from '@testing-library/react';
import * as React from 'react';
import { renderWithTheme } from 'test/renderUtils';

import CLIGuide from '..';
import CLIGuidesList from '../CLIGuidesList';

describe('CLIGuide', () => {
  it('renders without crashing', () => {
    const { container } = renderWithTheme(CLIGuide, {
      title: 'Get something from somewhere',
      children: '',
    });
    expect(container.firstChild).toMatchSnapshot();
  });

  it('renders the contents when open', async () => {
    const { container } = renderWithTheme(CLIGuide, {
      title: 'Get something from somewhere',
      children: 'Just some content',
    });

    fireEvent.click(
      screen.getByRole('button', { name: /Get something from somewhere/ })
    );

    expect(await screen.findByText('Just some content')).toBeInTheDocument();

    expect(container.firstChild).toMatchSnapshot();
  });
});

describe('CLIGuidesList', () => {
  it('renders without crashing', () => {
    const { container } = renderWithTheme(CLIGuidesList, {
      title: 'Use the API to …',
      children: '',
    });
    expect(container.firstChild).toMatchSnapshot();
  });

  it('renders the contents when open', async () => {
    const { container } = renderWithTheme(CLIGuidesList, {
      title: 'Use the API to …',
      children: [
        // eslint-disable-next-line react/jsx-key
        <CLIGuide title='Get something from somewhere' />,
        // eslint-disable-next-line react/jsx-key
        <CLIGuide title='Get something else from somewhere else' />,
      ],
    });

    fireEvent.click(screen.getByRole('button', { name: /Use the API to …/ }));

    expect(
      await screen.findByText('Get something from somewhere')
    ).toBeInTheDocument();
    expect(
      await screen.findByText('Get something else from somewhere else')
    ).toBeInTheDocument();

    expect(container.firstChild).toMatchSnapshot();
  });

  it(`renders the children as-is if there is only one`, () => {
    const { container } = renderWithTheme(CLIGuidesList, {
      title: 'Use the API to …',
      children: (
        <>
          {null}
          <CLIGuide title='Get something from somewhere' />
          {false}
        </>
      ),
    });

    expect(
      screen.queryByRole('button', { name: /Use the API to …/ })
    ).not.toBeInTheDocument();
    expect(container.firstChild).toMatchSnapshot();
  });
});
