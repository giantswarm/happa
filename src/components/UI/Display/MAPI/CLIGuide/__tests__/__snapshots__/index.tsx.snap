// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CLIGuide renders the contents when open 1`] = `
<div
  class="StyledGrommet-sc-19lkkz7-0 hRVLan"
>
  <div
    class="ThemeProvider__AppWrapper-sc-12xd2cf-0 eKUXDU"
  >
    <div
      class="StyledBox-sc-13pk1d4-0 ljkxXp CLIGuide__StyledAccordion-sc-1q6mc5x-0 hTUAzi"
    >
      <div
        class="StyledBox-sc-13pk1d4-0 dxLTUa"
      >
        <button
          aria-expanded="true"
          class="StyledButtonKind-sc-1vhfpnt-0 tWawK"
          type="button"
        >
          <div
            class="StyledBox-sc-13pk1d4-0 kEUDyz"
          >
            <div
              class="StyledBox-sc-13pk1d4-0 dfYTAc"
            >
              <span
                aria-hidden="true"
                class="StyledText-sc-1sadyjn-0 kUTnPv fa fa-shell"
                role="presentation"
              />
              <div
                class="StyledBox__StyledBoxGap-sc-13pk1d4-1 gcrEaW"
              />
              <span
                class="StyledText-sc-1sadyjn-0 kdtqol"
              >
                Get something from somewhere via the Management API
              </span>
            </div>
            <div
              class="StyledBox-sc-13pk1d4-0 gHlpEJ"
            >
              <svg
                aria-label="FormUp"
                class="StyledIcon-sc-ofa7kd-0 gLGkbm"
                viewBox="0 0 24 24"
              >
                <path
                  d="m18 15-6-6-6 6"
                  fill="none"
                  stroke="#000"
                  stroke-width="2"
                />
              </svg>
            </div>
          </div>
        </button>
        <div
          class="StyledBox-sc-13pk1d4-0 uHeKH"
          role="region"
          style=""
        >
          <div
            aria-hidden="false"
            class="StyledBox-sc-13pk1d4-0 uHeKH Collapsible__AnimatedBox-sc-15kniua-0 czoCTd"
            open=""
            style="max-height: 0;"
          >
            <div
              class="StyledBox-sc-13pk1d4-0 fMLcNm"
            >
              Just some content
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`CLIGuide renders without crashing 1`] = `
<div
  class="StyledGrommet-sc-19lkkz7-0 hRVLan"
>
  <div
    class="ThemeProvider__AppWrapper-sc-12xd2cf-0 eKUXDU"
  >
    <div
      class="StyledBox-sc-13pk1d4-0 ljkxXp CLIGuide__StyledAccordion-sc-1q6mc5x-0 hTUAzi"
    >
      <div
        class="StyledBox-sc-13pk1d4-0 dxLTUa"
      >
        <button
          aria-expanded="false"
          class="StyledButtonKind-sc-1vhfpnt-0 tWawK"
          type="button"
        >
          <div
            class="StyledBox-sc-13pk1d4-0 kEUDyz"
          >
            <div
              class="StyledBox-sc-13pk1d4-0 dfYTAc"
            >
              <span
                aria-hidden="true"
                class="StyledText-sc-1sadyjn-0 kUTnPv fa fa-shell"
                role="presentation"
              />
              <div
                class="StyledBox__StyledBoxGap-sc-13pk1d4-1 gcrEaW"
              />
              <span
                class="StyledText-sc-1sadyjn-0 kdtqol"
              >
                Get something from somewhere via the Management API
              </span>
            </div>
            <div
              class="StyledBox-sc-13pk1d4-0 gHlpEJ"
            >
              <svg
                aria-label="FormDown"
                class="StyledIcon-sc-ofa7kd-0 gLGkbm"
                viewBox="0 0 24 24"
              >
                <path
                  d="m18 9-6 6-6-6"
                  fill="none"
                  stroke="#000"
                  stroke-width="2"
                />
              </svg>
            </div>
          </div>
        </button>
        <div
          class="StyledBox-sc-13pk1d4-0 uHeKH"
          role="region"
        >
          <div
            aria-hidden="true"
            class="StyledBox-sc-13pk1d4-0 uHeKH Collapsible__AnimatedBox-sc-15kniua-0 iPDbNQ"
          />
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`CLIGuidesList renders the children as-is if there is only one 1`] = `
<div
  class="StyledGrommet-sc-19lkkz7-0 hRVLan"
>
  <div
    class="ThemeProvider__AppWrapper-sc-12xd2cf-0 eKUXDU"
  >
    <div
      class="StyledBox-sc-13pk1d4-0 uHeKH"
    >
      <div
        class="StyledBox-sc-13pk1d4-0 ljkxXp CLIGuide__StyledAccordion-sc-1q6mc5x-0 hTUAzi"
      >
        <div
          class="StyledBox-sc-13pk1d4-0 dxLTUa"
        >
          <button
            aria-expanded="false"
            class="StyledButtonKind-sc-1vhfpnt-0 tWawK"
            type="button"
          >
            <div
              class="StyledBox-sc-13pk1d4-0 kEUDyz"
            >
              <div
                class="StyledBox-sc-13pk1d4-0 dfYTAc"
              >
                <span
                  aria-hidden="true"
                  class="StyledText-sc-1sadyjn-0 kUTnPv fa fa-shell"
                  role="presentation"
                />
                <div
                  class="StyledBox__StyledBoxGap-sc-13pk1d4-1 gcrEaW"
                />
                <span
                  class="StyledText-sc-1sadyjn-0 kdtqol"
                >
                  Get something from somewhere via the Management API
                </span>
              </div>
              <div
                class="StyledBox-sc-13pk1d4-0 gHlpEJ"
              >
                <svg
                  aria-label="FormDown"
                  class="StyledIcon-sc-ofa7kd-0 gLGkbm"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="m18 9-6 6-6-6"
                    fill="none"
                    stroke="#000"
                    stroke-width="2"
                  />
                </svg>
              </div>
            </div>
          </button>
          <div
            class="StyledBox-sc-13pk1d4-0 uHeKH"
            role="region"
          >
            <div
              aria-hidden="true"
              class="StyledBox-sc-13pk1d4-0 uHeKH Collapsible__AnimatedBox-sc-15kniua-0 iPDbNQ"
            />
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`CLIGuidesList renders the contents when open 1`] = `
<div
  class="StyledGrommet-sc-19lkkz7-0 hRVLan"
>
  <div
    class="ThemeProvider__AppWrapper-sc-12xd2cf-0 eKUXDU"
  >
    <div
      class="StyledBox-sc-13pk1d4-0 bVXAXu"
    >
      <div
        class="StyledBox-sc-13pk1d4-0 dxLTUa"
      >
        <button
          aria-expanded="true"
          class="StyledButtonKind-sc-1vhfpnt-0 tWawK"
          type="button"
        >
          <div
            class="StyledBox-sc-13pk1d4-0 kEUDyz"
          >
            <div
              class="StyledBox-sc-13pk1d4-0 bqhqDO"
            >
              <span
                aria-hidden="true"
                class="StyledText-sc-1sadyjn-0 julYLc fa fa-shell"
                role="presentation"
              />
              <div
                class="StyledBox__StyledBoxGap-sc-13pk1d4-1 gcrEaW"
              />
              <span
                class="StyledText-sc-1sadyjn-0 knielG"
              >
                Use the API to â€¦
              </span>
            </div>
            <div
              class="StyledBox-sc-13pk1d4-0 gHlpEJ"
            >
              <svg
                aria-label="FormUp"
                class="StyledIcon-sc-ofa7kd-0 gLGkbm"
                viewBox="0 0 24 24"
              >
                <path
                  d="m18 15-6-6-6 6"
                  fill="none"
                  stroke="#000"
                  stroke-width="2"
                />
              </svg>
            </div>
          </div>
        </button>
        <div
          class="StyledBox-sc-13pk1d4-0 uHeKH"
          role="region"
          style=""
        >
          <div
            aria-hidden="false"
            class="StyledBox-sc-13pk1d4-0 uHeKH Collapsible__AnimatedBox-sc-15kniua-0 czoCTd"
            open=""
            style="max-height: 0;"
          >
            <div
              class="StyledBox-sc-13pk1d4-0 dNxRJm"
            >
              <div
                class="StyledBox-sc-13pk1d4-0 ljkxXp CLIGuide__StyledAccordion-sc-1q6mc5x-0 hTUAzi"
              >
                <div
                  class="StyledBox-sc-13pk1d4-0 dxLTUa"
                >
                  <button
                    aria-expanded="false"
                    class="StyledButtonKind-sc-1vhfpnt-0 tWawK"
                    type="button"
                  >
                    <div
                      class="StyledBox-sc-13pk1d4-0 kEUDyz"
                    >
                      <div
                        class="StyledBox-sc-13pk1d4-0 dfYTAc"
                      >
                        <span
                          aria-hidden="true"
                          class="StyledText-sc-1sadyjn-0 kUTnPv fa fa-shell"
                          role="presentation"
                        />
                        <div
                          class="StyledBox__StyledBoxGap-sc-13pk1d4-1 gcrEaW"
                        />
                        <span
                          class="StyledText-sc-1sadyjn-0 kdtqol"
                        >
                          Get something from somewhere
                        </span>
                      </div>
                      <div
                        class="StyledBox-sc-13pk1d4-0 gHlpEJ"
                      >
                        <svg
                          aria-label="FormDown"
                          class="StyledIcon-sc-ofa7kd-0 gLGkbm"
                          viewBox="0 0 24 24"
                        >
                          <path
                            d="m18 9-6 6-6-6"
                            fill="none"
                            stroke="#000"
                            stroke-width="2"
                          />
                        </svg>
                      </div>
                    </div>
                  </button>
                  <div
                    class="StyledBox-sc-13pk1d4-0 uHeKH"
                    role="region"
                  >
                    <div
                      aria-hidden="true"
                      class="StyledBox-sc-13pk1d4-0 uHeKH Collapsible__AnimatedBox-sc-15kniua-0 iPDbNQ"
                    />
                  </div>
                </div>
              </div>
              <div
                class="StyledBox__StyledBoxGap-sc-13pk1d4-1 iptgRM"
              />
              <div
                class="StyledBox-sc-13pk1d4-0 ljkxXp CLIGuide__StyledAccordion-sc-1q6mc5x-0 hTUAzi"
              >
                <div
                  class="StyledBox-sc-13pk1d4-0 dxLTUa"
                >
                  <button
                    aria-expanded="false"
                    class="StyledButtonKind-sc-1vhfpnt-0 tWawK"
                    type="button"
                  >
                    <div
                      class="StyledBox-sc-13pk1d4-0 kEUDyz"
                    >
                      <div
                        class="StyledBox-sc-13pk1d4-0 dfYTAc"
                      >
                        <span
                          aria-hidden="true"
                          class="StyledText-sc-1sadyjn-0 kUTnPv fa fa-shell"
                          role="presentation"
                        />
                        <div
                          class="StyledBox__StyledBoxGap-sc-13pk1d4-1 gcrEaW"
                        />
                        <span
                          class="StyledText-sc-1sadyjn-0 kdtqol"
                        >
                          Get something else from somewhere else
                        </span>
                      </div>
                      <div
                        class="StyledBox-sc-13pk1d4-0 gHlpEJ"
                      >
                        <svg
                          aria-label="FormDown"
                          class="StyledIcon-sc-ofa7kd-0 gLGkbm"
                          viewBox="0 0 24 24"
                        >
                          <path
                            d="m18 9-6 6-6-6"
                            fill="none"
                            stroke="#000"
                            stroke-width="2"
                          />
                        </svg>
                      </div>
                    </div>
                  </button>
                  <div
                    class="StyledBox-sc-13pk1d4-0 uHeKH"
                    role="region"
                  >
                    <div
                      aria-hidden="true"
                      class="StyledBox-sc-13pk1d4-0 uHeKH Collapsible__AnimatedBox-sc-15kniua-0 iPDbNQ"
                    />
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`CLIGuidesList renders without crashing 1`] = `
<div
  class="StyledGrommet-sc-19lkkz7-0 hRVLan"
>
  <div
    class="ThemeProvider__AppWrapper-sc-12xd2cf-0 eKUXDU"
  >
    <div
      class="StyledBox-sc-13pk1d4-0 uHeKH"
    />
  </div>
</div>
`;
